src/Language/PureScript/Make.hs

```diff
@@ -75,9 +76,19 @@ rebuildModule' MakeActions{..} exEnv externs m@(Module _ _ moduleName _ _) = do
   let env = foldl' (flip applyExternsFileToEnvironment) initEnvironment externs
       withPrim = importPrim m
   lint withPrim
+
   ((Module ss coms _ elaborated exps, env'), nextVar) <- runSupplyT 0 $ do
-    desugar exEnv externs [withPrim] >>= \case
-      [desugared] -> runCheck' (emptyCheckState env) $ typeCheckModule desugared
+    runStateT (desugar externs [withPrim]) (exEnv, mempty) >>= \case
+      ([desugared], (exEnv', usedImportsByModuleName)) -> do
+        (checked, CheckState{..}) <- runStateT (typeCheckModule desugared) $ emptyCheckState env
+        let usedImports = fold $ M.lookup moduleName usedImportsByModuleName
+            usedImports' = foldl' (flip $ \(fromModuleName, newtypeCtorName) ->
+              M.alter (Just . (fmap DctorName newtypeCtorName :) . fold) fromModuleName) usedImports checkCoercedNewtypeCtorsImports
+        -- Imports cannot be linted before type checking because we need to
+        -- known which newtype constructors are used to solve Coercible
+        -- constraints in order to not report them as unused.
+        censor (addHint (ErrorInule moduleName)) $ lintImports checked exEnv' usedImports'
```

Comment:
typo here? I think this is causing CI to fail

Reply:
I guess I was too tired even for adding comments properly 🤦